lib/repository.dart

```diff
@@ -180,7 +180,7 @@ class BluetoothRepository {
     return null;
   }
 
-  Future<List<int>?> readCurrentState(String deviceId) async {
+  Future<List<int>?> readCurrentState(String deviceId, currentStateId) async {
```

Comment:
Again, I think this should just read all the data in one go, returning a `Map<String, List<int>>`.

Reply:
I'm worried, if we read all the data in one go. 
The function will start to bottleneck, as its waiting to get all the data as opposed to the specific data I need.

Reply:
@AxelFougues 
I think Walker73 is getting all the data at once,
Does it causes any issues?

Reply:
What data are we talking about? 
W73 reads all the readable IDs on connexion but it does so iteratively :
Send request
Wait for answer
Repeat

Not waiting for an answer between reads did cause it to skip some replies.

Then it subscribes to the notifications.

Reply:
Thanks Az.

@blopker, just confirming do you want this change in this PR or next PR?

Reply:
What I was mostly suggesting is just moving the code from here into `readCurrentState`, but awaiting each actual read. It shouldn't have any functional difference, it's mostly about encapsulation of the low level BT internals.

Reply:
Understood.
Will do it in next PR, when I investigate the subscription method and get my bike upstairs again.